services/payments/datastore.go

```diff
@@ -43,13 +113,37 @@ var (
 )
 
 // SignTransaction - perform KMS signing of the transaction, return publicKey and signature in hex string
-func (t *Transaction) SignTransaction(ctx context.Context) (string, string, error) {
-	// TODO: fill in
-	return "", "", errorutils.ErrNotImplemented
+func (t *Transaction) SignTransaction(ctx context.Context, kmsClient wrappedKMSClient, keyId string) (string, string, error) {
+	pubkeyOutput, err := kmsClient.GetPublicKey(ctx, &kms.GetPublicKeyInput{
+		KeyId: &keyId,
+	})
+
+	if err != nil {
+		return "", "", fmt.Errorf("Failed to get public key: %w", err)
+	}
+
+	signingOutput, err := kmsClient.Sign(ctx, &kms.SignInput{
+		KeyId:            &keyId,
+		Message:          t.BuildSigningBytes(),
+		SigningAlgorithm: kmsTypes.SigningAlgorithmSpecEcdsaSha256,
+	})
+
+	if err != nil {
+		return "", "", fmt.Errorf("Failed to sign transaction: %w", err)
+	}
+
+	return hex.EncodeToString(pubkeyOutput.PublicKey), hex.EncodeToString(signingOutput.Signature), nil
+}
+
+// BuildSigningBytes - the string format that payments will sign over per tx
+func (t *Transaction) BuildSigningBytes() []byte {
```

Comment:
This is not the direction we chose to go in, signing bytes is the json blob of the transaction which is stored in qldb, not generated by attributes of a transaction anymore.

Reply:
Yeah this function is vestigial and I took it out. I want to discuss using `ion.MarshalBinary` instead of JSON for serializing into QLDB, but that's in addition to dropping this function.